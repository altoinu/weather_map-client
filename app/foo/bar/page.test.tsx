import Page from "./page";
import { render, screen } from "@testing-library/react";
import { usePathname } from "next/navigation";

jest.mock("next/navigation");

const mockUsePathname = usePathname as jest.Mock;

describe("bar page", () => {
  beforeAll(() => {
    mockUsePathname.mockReturnValue("/mockURL");
  });

  it("should render", () => {
    const { container: pageContainer } = render(<Page />);

    expect(pageContainer).toMatchSnapshot();
  });

  it("should render a heading", () => {
    render(<Page />);
    const heading = screen.getByRole("heading");
    expect(heading.textContent).toBe("This is Bar page.");
  });
});
