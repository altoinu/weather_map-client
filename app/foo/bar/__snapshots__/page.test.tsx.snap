// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bar page should render 1`] = `
<div>
  <h2>
    This is Bar page.
  </h2>
  <p>
    Pathname: 
    /mockURL
  </p>
  <ul>
    <li>
      <a
        href="/mockURL/.."
        style="color: red; text-decoration: underline;"
      >
        return to one level above page
      </a>
    </li>
    <li>
      <a
        href="/mockURL/async/hello"
        style="color: red; text-decoration: underline;"
      >
        to asynchronous [slug] page
      </a>
    </li>
    <li>
      <a
        href="/mockURL/sync/hello"
        style="color: red; text-decoration: underline;"
      >
        to synchronous [slug] page
      </a>
    </li>
  </ul>
  <br />
</div>
`;
